stream/src/main/java/org/apache/kafka/streaming/examples/ProcessorJob.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.kafka.streaming.KafkaStreaming;
+import org.apache.kafka.streaming.processor.Processor;
+import org.apache.kafka.streaming.processor.ProcessorDef;
+import org.apache.kafka.streaming.processor.TopologyBuilder;
+import org.apache.kafka.streaming.processor.ProcessorContext;
+import org.apache.kafka.streaming.StreamingConfig;
+import org.apache.kafka.streaming.state.Entry;
+import org.apache.kafka.streaming.state.InMemoryKeyValueStore;
+import org.apache.kafka.streaming.state.KeyValueIterator;
+import org.apache.kafka.streaming.state.KeyValueStore;
+
+import java.util.Properties;
+
+public class ProcessorJob {
+
+    private static class MyProcessorDef implements ProcessorDef {
+
+        @Override
+        public Processor<String, Integer> instance() {
+            return new Processor<String, Integer>() {
+                private ProcessorContext context;
+                private KeyValueStore<String, Integer> kvStore;
+
+                @Override
+                public void init(ProcessorContext context) {
+                    this.context = context;
+                    this.context.schedule(this, 1000);
+                    this.kvStore = new InMemoryKeyValueStore<>("local-state", context);
+                }
+
+                @Override
+                public void process(String key, Integer value) {
+                    Integer oldValue = this.kvStore.get(key);
+                    if (oldValue == null) {
+                        this.kvStore.put(key, value);
+                    } else {
+                        int newValue = oldValue + value;
+                        this.kvStore.put(key, newValue);
+                    }
+
+                    context.commit();
+                }
+
+                @Override
+                public void punctuate(long streamTime) {
+                    KeyValueIterator<String, Integer> iter = this.kvStore.all();
+
+                    while (iter.hasNext()) {
+                        Entry<String, Integer> entry = iter.next();
+
+                        System.out.println("[" + entry.key() + ", " + entry.value() + "]");
+
+                        context.forward(entry.key(), entry.value());
+                    }
+                }
+
+                @Override
+                public void close() {
+                    this.kvStore.close();
+                }
+            };
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        StreamingConfig config = new StreamingConfig(new Properties());
+        TopologyBuilder builder = new TopologyBuilder();
+
+        builder.addSource("SOURCE", new StringDeserializer(), new IntegerDeserializer(), "topic-source");
+
+        builder.addProcessor("PROCESS", new MyProcessorDef(), null, "SOURCE");
+
+        builder.addSink("SINK", "topic-sink", new StringSerializer(), new IntegerSerializer(), "PROCESS");
+
+        KafkaStreaming streaming = new KafkaStreaming(builder, config);
+        streaming.start();
```

Comment:
StreamingConfig is supposed to only include config values that are used by the KafkaStreaming runtime but not in the user logic, if users do want to modify the behavior of their processors based on some StreamingConfig values they can either do:

```
builder.addProcessor("PROCESS", new MyProcessorDef(config.get(CONFIG_NAME1), config.get(CONFIG_NAME2)), null, "SOURCE");
```

Or they can also pass-in the whole StreamingConfig object, which of course can be different from the one they passed into KafkaStreaming, into their instantiated ProcessorDef constructors, although I personally would not recommend this way. Generally I think StreamingConfig should not be exposed to the processor interface layer, since it is designed to be used only at the runtime level.